sx126x/sx126x.go

```diff
@@ -0,0 +1,690 @@
+	p[3] = lowDataRateOptimize
+	d.ExecSetCommand(SX126X_CMD_SET_MODULATION_PARAMS, p[:])
+}
+
+// CheckDeviceReady sleep until all busy flags clears
+func (d *Device) CheckDeviceReady() error {
+	if d.deepSleep == true {
+		d.SpiSetNss(false)
+		time.Sleep(time.Millisecond)
+		d.SpiSetNss(true)
+		d.deepSleep = false
+	}
+	return d.WaitBusy()
+}
+
+// ExecSetCommand send a command to configure the peripheral
+func (d *Device) ExecSetCommand(cmd uint8, buf []uint8) {
+	d.CheckDeviceReady()
+	if cmd == SX126X_CMD_SET_SLEEP {
+		d.deepSleep = true
+	} else {
+		d.deepSleep = false
+	}
+	d.SpiSetNss(false)
+	// Send command and params
+	d.SpiTx(cmd)
+	for i := 0; i < len(buf); i++ {
+		d.SpiTx(buf[i])
+	}
+	d.SpiSetNss(true)
+	if cmd != SX126X_CMD_SET_SLEEP {
+		d.WaitBusy()
+	}
+}
+
+// ExecGetCommand queries the peripheral the peripheral
+func (d *Device) ExecGetCommand(cmd uint8, size uint8) []uint8 {
+	d.CheckDeviceReady()
+	d.SpiSetNss(false)
+	// Send the command and flush first status byte (as not used)
+	d.SpiTx(cmd)
+	d.SpiTx(0x00)
+	var ret []uint8
+	// Receive given number of bytes from radio
+	for i := uint8(0); i < size; i++ {
+		ret = append(ret, d.SpiTx(0xFF))
+	}
+	d.SpiSetNss(true)
+	d.WaitBusy()
+	return ret
+}
+
+//
+// Configuration
+//
+
+// SetLoraFrequency() Sets current Lora Frequency
+// NB: Change will be applied at next RX / TX
+func (d *Device) SetLoraFrequency(freq uint32) {
+	d.loraConf.Freq = d.loraConf.Freq
+}
+
+// SetLoraIqMode() defines the current IQ Mode (Standard/Inverted)
+// NB: Change will be applied at next RX / TX
+func (d *Device) SetLoraIqMode(mode uint8) {
+	if mode == 0 {
+		d.loraConf.Iq = SX126X_LORA_IQ_STANDARD
+	} else {
+		d.loraConf.Iq = SX126X_LORA_IQ_INVERTED
+	}
+}
+
+// SetLoraCodingRate() sets current Lora Coding Rate
+// NB: Change will be applied at next RX / TX
+func (d *Device) SetLoraCodingRate(cr uint8) {
+	d.loraConf.Cr = cr
+}
+
+// SetLoraBandwidth() sets current Lora Bandwidth
+// NB: Change will be applied at next RX / TX
+func (d *Device) SetLoraBandwidth(bw uint8) {
+	d.loraConf.Cr = bw
+}
+
+// SetLoraCrc() sets current CRC mode (ON/OFF)
+// NB: Change will be applied at next RX / TX
+func (d *Device) SetLoraCrc(enable bool) {
+	if enable {
+		d.loraConf.Crc = SX126X_LORA_CRC_ON
+	} else {
+		d.loraConf.Crc = SX126X_LORA_CRC_OFF
+	}
+}
+
+//SetLoraSpreadingFactor setc surrent Lora Spreading Factor
+// NB: Change will be applied at next RX / TX
+func (d *Device) SetLoraSpreadingFactor(sf uint8) {
+	d.loraConf.Sf = sf
+}
+
+//
+// Lora functions
+//
+//
+
+// LoraConfig() initializes the driver for Lora operations
+func (d *Device) LoraConfig(cnf LoraConfig) {
+
+	// Save given configuration
+	d.loraConf = cnf
+
+	// Switch to standby prior to configuration changes
+	d.SetStandby()
+
+	// Clear errors, disable radio interrupts for the moment
+	d.ClearDeviceErrors()
+	d.ClearIrqStatus(SX126X_IRQ_ALL)
+	d.SetDioIrqParams(0x00, 0x00, 0x00, 0x00)
+
+	// radio.CalibrateAll()
```

Comment:
Remove commented?

Reply:
Done. 
Checked for other useless comments in the whole driver